### PR TITLE
`EcDoubleSlopeV3` + `ComputeSlopeV3` hints

### LINKED ISSUE #535: `EcDoubleSlopeV3` hint (secp hint)
https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/cairo_programs/ed25519_ec.cairo#L27

https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/security/whitelists/cairo_secp.json#L188

https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/vm/src/hint_processor/builtin_hint_processor/hint_code.rs#L689C11-L689C29

### LINKED ISSUE #536: `ComputeSlopeV3` hint (secp hint)
https://github.com/starkware-libs/cairo-lang/blob/0e4dab8a6065d80d1c726394f5d9d23cb451706a/src/starkware/starknet/security/whitelists/cairo_secp.json#L200

https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/cairo_programs/ed25519_ec.cairo#L57

https://github.com/lambdaclass/cairo-vm/blob/db7fff18c9c1312024ebf4119fcdfeba23657b12/vm/src/hint_processor/builtin_hint_processor/hint_code.rs#L758

### PR DESCRIPTION
**Pull request by TAdev0**
Resolves #535 
Resolves #536 

ready for review @har777  @cicr99 , integration tests pass

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 1]
```diff
@@ -390,6 +390,84 @@ func createEcDoubleSlopeV1Hinter(resolver hintReferenceResolver) (hinter.Hinter,
 	return newEcDoubleSlopeV1Hint(point), nil
 }
 
+// EcDoubleSlopeV3 hint computes the slope for doubling a point on the elliptic curve
+//
+// `newEcDoubleSlopeV3Hint` takes 1 operander as argument
+//   - `pt` is the point on an elliptic curve to operate on
+//
+// `newEcDoubleSlopeV3Hint` assigns the `slope` result as `value` in the current scope
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_ec.go:398]: Could you add a comment here explaining the difference between versions?

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 2]
```diff
@@ -715,6 +793,126 @@ func createComputeSlopeV1Hinter(resolver hintReferenceResolver) (hinter.Hinter,
 	return newComputeSlopeV1Hint(point0, point1), nil
 }
 
+// ComputeSlopeV3 hint computes the slope between two points on an elliptic curve
+//
+// `newComputeSlopeV3Hint` takes 2 operanders as arguments
+//   - `pt0` is the first point on an elliptic curve to operate on
+//   - `pt1` is the second point on an elliptic curve to operate on
+//
+// `newComputeSlopeV3Hint` assigns the `slope` result as `value` in the current scope
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_ec.go:802]: Same comment here for these hint pls

File: pkg/hintrunner/zero/zerohint_ec.go
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_ec.go:1]: Should we add some unit tests for these new versions? I see the way the value is calculated is what changed.

File: pkg/hintrunner/zero/zerohint_ec.go (hunk 1 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_ec.go:460]: done!

File: pkg/hintrunner/zero/zerohint_ec.go (hunk 2 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_ec.go:944]: done!

File: pkg/hintrunner/zero/zerohint_ec.go
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_ec.go:1]: in both case computation is strictly the same
just V1 uses a utility and adds a check compared to V3

see for example the python utility :

```
def line_slope(point1: Tuple[int, int], point2: Tuple[int, int], p: int) -> int:
    """
    Computes the slope of the line connecting the two given EC points over the field GF(p).
    Assumes the points are given in affine form (x, y) and have different x coordinates.
    """
    assert (point1[0] - point2[0]) % p != 0
    return div_mod(point1[1] - point2[1], point1[0] - point2[0], p)
  ```

used in `ComputeSlopeV1` : 
```
value = slope = line_slope(point1=(x0, y0), point2=(x1, y1), p=SECP_P)
```

with `ComputeSlopeV3` being:

```
 value = slope = div_mod(y0 - y1, x0 - x1, SECP_P)
````

**Comment by TAdev0** [comment]
@cicr99  just added unit tests, but i think its a better idea to refactor all this as there is common code , with only one unit tests for the computation being done
will do that now

File: pkg/hintrunner/zero/zerohint_ec.go
Comment (cicr99) [reply pkg/hintrunner/zero/zerohint_ec.go:1]: Oh I see what you mean. Then you are right, no need to add new tests then

**Review by cicr99 — APPROVED** [review]

**Pull request merged**